### ISSUE TITLE
regression control, lost the ability to use a ratio anchor.

### ISSUE THREAD

**Issue by OlexiyKravchuk** [godotengine/godot#27356]
**Godot version:**
Godot_v3.1-stable_win64.exe

**Issue description:**
I began to transfer my projects from 2.1.5 to 3.1 and found that there is no way to scale my interface, without shaman dances.

**Steps to reproduce:**
Go to the editor, put the control and try to set the anchor in the ratio mode.

**Comment by Calinou** [comment]
What do you mean by radial anchor? Can you take a screenshot of what you were using in 2.1?

**Comment by OlexiyKravchuk** [comment]
> Calinou

![image](https://user-images.githubusercontent.com/10380658/54869953-8a33b600-4da8-11e9-8c27-85917735906b.png)

**Comment by OlexiyKravchuk** [comment]
sorry there was a typo

**Comment by Zylann** [comment]
`Begin` was actually `0.0`, `Center` was `0.5`, `End` was `1.0`, and `Ratio` was a way to set that number to what you want.
Now it actually is a ratio by default, you can set those numbers here:
![image](https://user-images.githubusercontent.com/1311555/54870246-e7753a80-4d9b-11e9-89cb-a34368881078.png)

**Comment by groud** [comment]
Yes, as Zylann this is perfectly possible in 3.1. If you need help on that, please ask on the other community channels.

**Comment by OlexiyKravchuk** [comment]
I did not understand how everything about what you say solves this problem?

**Comment by OlexiyKravchuk** [comment]
that example that you gave is absolutely not suitable, instead of setting one value in the field and forgetting about the problems, I need to reinvent the "wheel", which will essentially correct any changes in proportions manually and recalculate the relative position and scale of each element , this not only adds me a lot of work, but also lowers the program's performance, why in general someone thought up this? it's the worst design of all possible!

**Comment by groud** [comment]
No, you just have to use the anchors values that basically correspond to the same ratio value you are talking about. Anchors are relative to the parent size, while margins are absolute pixels values. By combining them your can achieve any dynamic placement you need.

**Comment by Zylann** [comment]
I think it's the fact Godot 2.1.5 supported ratio-based control rects. When you changed the rectangle in the editor, it was updating the ratios. But in 3.x, it modifies margins instead, which is why it now requires to set ratios manually rather than visually. Is that it, @OlexiyKravchuk ?

**Comment by OlexiyKravchuk** [comment]
I created a small demo project for version 2.1.5 to explain what effect I need. In it, you can stretch the window with the background image as you like, but the buttons on the eyes will always be in place observing the proportion and position. To do this, I just set the buttons in the right place and set the ratio mode in the anchor fields.
What fields should I change to achieve this result in 3.1? If possible a minimal example of the same project.
[ratio_anchor.zip](https://github.com/godotengine/godot/files/2999861/ratio_anchor.zip)

**Comment by groud** [comment]
Ah, you need to position the anchors at the corner of the node itself. You can easily achieve that by displaying the anchors helpers (in the view menu) and drag them to the corners of the node. Use the smart snapping to do so.
It's a little bit cumbersome to do for now, but this is already covered by another issue.

**Comment by groud** [comment]
See #15252

**Comment by OlexiyKravchuk** [comment]
Hmm ... sorry, but I absolutely did not understand what to do.
P.S. I also cannot understand why it was necessary to change the elegant and effective system of ratio anchors that worked well, for something like that, in which was not possible to work even with hints.

**Comment by OlexiyKravchuk** [comment]
I found such a [cheat sheet](https://docs.godotengine.org/en/3.1/getting_started/step_by_step/ui_introduction_to_the_ui_system.html#anchors-are-relative-to-the-parent-container) in the documentation, but even following it and your advice, it is impossible to reproduce the same result as in 2.1.5 due to the fact that the green pins in 3.1 cannot be moved beyond the boundaries of the parent node, while in 2.1. 5 the system works like a clock, regardless of the hierarchy and positioning of the child nodes.
Here is [ratio_anchor_v3.1.zip](https://github.com/godotengine/godot/files/2999949/ratio_anchor_v3.1.zip) an example of a project in 3.1, in it only the first button is able to work as it should.
In addition, I suppose if during the execution you change the position of the nodes, then after that, the scaling and offset recalculation will occur relative to the old manual coordinates, and this is a new problem.

**Comment by groud** [comment]
Yep, someone complained about that in #21080.

**Comment by OlexiyKravchuk** [comment]
I read it and didn’t understand why it couldn’t be done so that it was comfortable without nails in the ass. It’s simply stated that it’s impossible, but no objective reasons have been given. And instead of solving many recommendations aimed at the invention of the wheels, thousands of times in a row, for each case.
  (maybe I didn’t understand something, but I think there are hidden personal motives that prevent a rational and correct decision from being made)

**Comment by groud** [comment]
Please, you also have to understand that your use case is not the only one regarding control node. The previous solution was a stack of inconsistent workarounds that did not help people understand how the margin and archors system worked.

To solve the problem, I proposed a new, more consistent design, that was reviewed and accepted by the community. Indeed, it has some drawbacks as pointed out by the mentioned issues and that could be solved, that's why those issues are still open. The use case you are talking about is indeed one important problem, as, I agree, it is not easy right now to keep a control node as a ratio of its parent.

Also, I don't really get why you would think that I, or anyone who contributed to the anchor and margin system, would have personal interest in not not taking rational decision. Building such system in a consistent way is not easy, and for now we have a lot less issue with the new system than before.

But if you feel like you can come up with a smarter design, feel free to propose your own changes by opening a PR ;)

**Comment by OlexiyKravchuk** [comment]
I'll be honest. I do not know what it was under the hood before.
But I clearly see that earlier I was able to figure out what was happening, within a couple of minutes, and I could solve almost any problem with the interface myself. And now I have already departed for more than a week of time, in order to solve a problem of which I did not have before, and at the moment it turned out that it would not be possible to solve it.
